### PR TITLE
misc(core): make ingestion-memory common pool on a node

### PR DESCRIPTION
**Pull request by sherali42**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?

- IngestionMemory is partitioned between datasets, causing sub-optimal utilization of memory. With this change, all configured datasets use single pool of memory.

### CODE REVIEW

File: core/src/main/scala/filodb.core/downsample/OffHeapMemory.scala [hunk 1]
```diff
@@ -14,7 +15,7 @@ class OffHeapMemory(schemas: Seq[Schema],
   extends StrictLogging {
 
   private val blockMemSize = storeConfig.shardMemSize
-  private val nativeMemSize = storeConfig.ingestionBufferMemSize
+  private val nativeMemSize = systemConfig.getMemorySize("filodb.memstore.ingestion-buffer-mem-size").toBytes
```
Comment (vishramachandran) [inline core/src/main/scala/filodb.core/downsample/OffHeapMemory.scala:18]: Decide whether you want to use same config. We could also use a different config for downsampler. I seem to prefer latter. If we go this route, previous comment is canceled.

**Review by vishramachandran — APPROVED** [review]
Approved with minor comments.

File: coordinator/src/test/scala/filodb.coordinator/IngestionStreamSpec.scala [hunk 2]
```diff
@@ -31,7 +31,10 @@ class IngestionStreamSpec extends ActorTest(IngestionStreamSpec.getNewSystem) wi
   implicit val defaultPatience =
     PatienceConfig(timeout = Span(10, Seconds), interval = Span(50, Millis))
 
-  val config = ConfigFactory.parseString("""filodb.memstore.groups-per-shard = 4""".stripMargin)
+  val config = ConfigFactory.parseString(
+    """filodb.memstore.groups-per-shard = 4
+      |filodb.memstore.ingestion-buffer-mem-size = 50MB
+      |""".stripMargin)
```
[the remark this answers is not in the document]
Comment (vishramachandran) [reply coordinator/src/test/scala/filodb.coordinator/IngestionStreamSpec.scala:37]: I see this already. Why override ?

File: core/src/main/resources/filodb-defaults.conf [hunk 3]
```diff
@@ -412,6 +408,10 @@ filodb {
     # It should be large enough to accommodate a new partition key, a new OffheapLFSortedIDMap structure
     # (a couple hundred bytes) plus 1000 write buffers.
     min-write-buffers-free = 5MB
+
+    # Number of bytes of offheap mem to allocate to write buffers for all shards.  Ex. 1000MB, 1G, 2GB
+    # Note: this memory is shared across all configued datasets on a node.
+    ingestion-buffer-mem-size = 200MB
```
[the remark this answers is not in the document]
Comment (sherali42) [reply core/src/main/resources/filodb-defaults.conf:414]: @vishramachandran as discussed this config is not applicable for downsample cluster

File: core/src/main/scala/filodb.core/downsample/OffHeapMemory.scala (hunk 1 above)
Comment (sherali42) [reply core/src/main/scala/filodb.core/downsample/OffHeapMemory.scala:18]: @vishramachandran as discussed this config is not applicable for downsample cluster

File: coordinator/src/test/scala/filodb.coordinator/IngestionStreamSpec.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (sherali42) [reply coordinator/src/test/scala/filodb.coordinator/IngestionStreamSpec.scala:37]: default is 200MB, and it is being overridden to 50MB in the existing code.

**Pull request merged**